x-pack/packages/ml/url_state/src/url_state.test.tsx

```diff
@@ -61,29 +50,96 @@ describe('getUrlState', () => {
 });
 
 describe('useUrlState', () => {
-  beforeEach(() => {
-    mockHistoryPush.mockClear();
+  it('pushes a properly encoded search string to history', () => {
+    const TestComponent: FC = () => {
```

Comment:
did you consider using the `renderHook` for tests instead of a component?

Reply:
I agree usually this would be the way to go, however in the test below the setup is necessary because we're testing the behaviour when two instances of the same hook are used and if they play nicely with the state stored in the URL which isn't possible with `renderHook` since it only supports one hook. I just copied the approach taken in the other test over to here so it's the same setup across the file.